client/src/main/java/org/asynchttpclient/netty/request/NettyRequestSender.java

```diff
@@ -265,16 +265,10 @@ private Channel getOpenChannel(NettyResponseFuture<?> future, Request request, P
 
         Object partitionKey = future.getPartitionKey();
 
-        // we disable channelPreemption when performing next requests
-        final boolean acquireChannelLock = !performingNextRequest;
```

Comment:
> The original author's intent was to not require an extra semaphore when being redirected to another host.

And that is actually wrong IMHO. For example, because of that behaviour you can run off your operating system file descriptor limit which could be configured to be almost equal to connection count limit.